CODE-OF-CONDUCT.md

```diff
@@ -0,0 +1,132 @@
+# Istio Community Code of Conduct
+
+All members of the Istio community must abide by the following Code of Conduct,
+adapted from the [Contributor Covenant][homepage]. Only by respecting each
+other can we develop a productive, collaborative community.
+
+## Our Pledge
+
+As members, contributors, and leaders of the Istio community, we pledge to make
+participation in our community a harassment-free experience for everyone,
+regardless of age, body size, visible or invisible disability, ethnicity, sex
+characteristics, gender identity and expression, level of experience,
+education, socio-economic status, nationality, personal appearance, race,
+religion, or sexual identity and orientation.
+
+We pledge to act and interact in ways that contribute to an open, welcoming,
+diverse, inclusive, and healthy community.
+
+## Our Standards
+
+Examples of behavior that contributes to a positive environment for our
+community include:
+
+* Demonstrating empathy and kindness toward other people
+* Being respectful of differing opinions, viewpoints, and experiences
+* Giving and gracefully accepting constructive feedback
+* Accepting responsibility and apologizing to those affected by our mistakes,
+  and learning from the experience
+* Focusing on what is best not just for us as individuals, but for the
+  overall community
+
+Examples of unacceptable behavior include:
+
+* The use of sexualized language or imagery, and sexual attention or
+  advances of any kind
+* Trolling, insulting or derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or email
+  address, without their explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+  professional setting
+
+## Enforcement Responsibilities
+
+Community leaders (including maintainers, Working Group leads, the Technical
+Oversight Committee and the Steering Committee) are responsible for clarifying
+and enforcing our standards of acceptable behavior and will take appropriate
+and fair corrective action in response to any behavior that they deem
+inappropriate, threatening, offensive, or harmful.
```

Comment:
Would be good to indicate that that the enforcement options that could be taken are listed under the Enforcement Guidelines.

Reply:
And additionally that the enforcement options are not sequential (i.e. we don't have to do option 1 before option 2). On that front, I think it might be helpful to remove the numbering on the enforcement guidelines entirely, so it's `### Correction`, `### Warning` rather than `### 1. Correction`, `### 2. Warning` etc.

Reply:
I think we're agreed this isn't a necessary change; having them numbered does not mean the enforcements are sequential.